docker/platypus-deps/Dockerfile

```diff
@@ -1,13 +1,154 @@
-# Build Platypus dependencies
 # Get base image
-FROM alexanderianblair/hephaestus-deps:master
+FROM ubuntu:22.04
 
-# By default four cores are used to compile
+# This is needed or it mpiexec complains because docker runs as root
+# Discussion on this issue https://github.com/open-mpi/ompi/issues/4451
+ENV OMPI_ALLOW_RUN_AS_ROOT=1
+ENV OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
+
+# By default one core is used to compile
 ARG compile_cores=4
 
-# By default we install everything under /opt
+# By default checkout mfem-moose branch
 ARG WORKDIR=opt
 
+# Install apt dependencies
+# hadolint ignore=DL3008
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get install -y --no-install-recommends \
+    build-essential \
+    bison \
+    flex \
+    clang-format \
+    clang-tidy \
+    curl \
+    doxygen \
+    git \
+    ninja-build \
+    pre-commit \
+    python3 \
+    python3-dev \
+    python3-distutils \
+    python-is-python3 \
+    python3-pip \
+    libboost-all-dev \
+    libfontconfig1-dev \
+    libfreetype6-dev \
+    libglew-dev \
+    libglm-dev \
+    libhdf5-mpich-dev \
+    libnetcdf-dev \
+    libomp-14-dev \
+    libpng-dev \
+    libsdl2-dev \
+    libssl-dev \
+    rsync \
+    xxd && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install pip packages
+# hadolint ignore=DL3042,DL3013
+RUN pip install packaging && \
+    pip install pyyaml && \
+    pip install setuptools && \
+    pip install jinja2
+
+# Install CMake
+WORKDIR /$WORKDIR
+# hadolint ignore=DL3003
+RUN git clone https://github.com/Kitware/CMake.git && \
+    cd CMake/ && \
+    git checkout 78ae759 && \
+    ./bootstrap --parallel="$compile_cores" && \
+    make -j"$compile_cores" && \
+    make install && \
+    alias cmake='/$WORKDIR/CMake/bin/cmake'
+
+# Download SuperLU
+WORKDIR /$WORKDIR
+RUN curl -LJO https://github.com/xiaoyeli/superlu_dist/archive/refs/tags/v8.1.0.tar.gz
+
+# Download PETSc
+WORKDIR /$WORKDIR
+# hadolint ignore=DL3003,SC2006
```

Comment:
Ideally we wouldn't add all these ignore statements to silence linting checks - would be better to follow them